counsel.el

```diff
@@ -3219,10 +3256,12 @@ substituted by the search regexp and file, respectively.  Neither
   (or
    (ivy-more-chars)
    (let* ((regex (counsel--grep-regex string))
-          (cmd (format counsel-grep-command (shell-quote-argument regex))))
+          (cmd (counsel--format
+                counsel-grep-command
+                (shell-quote-argument regex))))
      (counsel--async-command
       (if (ivy--case-fold-p regex)
-          (progn
+          (if (listp cmd) (append (list (car cmd) "-i") (cdr cmd))
```

Comment:
Nit: This `append` can be `nconc` because its first argument is a newly allocated list.